src/table_builder.cc

```diff
@@ -52,19 +53,19 @@ void TitanTableBuilder::Add(const Slice& key, const Slice& value) {
       base_builder_->Add(key, value);
     }
   } else if (ikey.type == kTypeValue &&
-             value.size() >= cf_options_.min_blob_size &&
```

Comment:
How about checking the state of BlobFileBuilder and whether is small kv in `AddToBaseTable` instead of in `AddBlob`. If in buffer state, you can append the small kv into `BlobFileBuilder` by another interface.

If that, you can remove the `HasFileWriter` of `BlobFileBuilder` and still keep the `value.size() >= cf_options_.min_blob_size` logic into `TableBuilder` which makes the logic tight and clear.

Reply:
Yeah, it would be clearer to check `BlobFileBuilder`'s state here. 
But still, the `HasFileWriter` is hard to remove, it is used to allow empty writer when `BlobFileBuilder` construct. Otherwise, we will have a needless blob file with no record when inserting small kv pairs with dictionary compression enabled.
check discussion here: https://tikv-wg.slack.com/archives/CQ545HDGC/p1602417406036300?thread_ts=1602417359.036200&cid=CQ545HDGC

Reply:
Check `BlobFileBuilder's state here, then there is no need to call `AddBlob` for small kv pairs so that there is no case for needless blob file created.

Reply:
> Check `BlobFileBuilder's state here, then there is no need to call `AddBlob` for small kv pairs so that there is no case for needless blob file created.

yeah, I see. I will refactor this later.

Reply:
updated